daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/BinaryNumberTraits.scala

```diff
@@ -50,3 +56,63 @@ trait HasRuntimeExplicitLength {
     nBytes * toBits
   }
 }
+
+
+trait PrefixedLengthParserMixin {
+
+  def prefixedLengthParser: Parser
+  def prefixedLengthERD: ElementRuntimeData
+  def lengthUnits: LengthUnits
+  def prefixedLengthAdjustmentInUnits: Long
```

Comment:
Don't think this can be a Long. I think it has to be an Evaluatable[Long], since the prefix length field itself can be variable length determined at runtime. 

I'm ok with putting that off to a future change. A useful comment about this limitation then would be all.

Reply:
The spec says:

> It is a schema definition error if the xs:simpleType [referenced by prefixLengthType] specifies any of:
> ...
> * dfdl:lengthKind 'explicit' where length is an expression
> ...

So I'm not sure it will ever need to be evaluatable. THe prefix length is always known at schema compile time. Except for the case in nested prefixes, but I think that wil result in lots more changes. I'm not sure how that would work, but I'm not sure an evaluatable will be right way to do it.

Reply:
Ok, looks like your code is correct. 

Hmm. I just reread the spec. It is narrower than I recall. It prohibits, for example this rep:

     123|...string of length 123 bytes...

Where the 123 is delimited by the "|". I believe I've actually seen this somewhere.
It does allow

    123...string of length 123...

where the 123 is a fixed-length text string of length 3 bytes. 

Based on what it says in the spec. the only way the prefix itself can be variable length is if it itself is this nested-prefix-length thing where the prefix length itself has  a prefix length. There's exactly one case of that in the world and it's related to an ASN.1 feature. 

I suspect this is a mistake in the spec. In general the prefix fields themselves can be variable length.
At least lengthKind 'delimited' is needed, and the special case of the prefix itself having a prefix is supposed to the "wierd thing" that can happen in addition to all the other ways the prefix itself could be variable length.

But that will have to get fixed in a rev of the DFDL spec. For now what you have seems to match the spec. so good enough.